### PR TITLE
Fixes Brave Ads assertion is thrown if ads service is not bound or initialized when getting transaction history

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/11411

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/ui/webui/brave_rewards_page_ui.cc [hunk 1]
```diff
@@ -1453,35 +1454,42 @@ void RewardsDOMHandler::OnPublisherListNormalized(
...
-          base::Time::FromDoubleT(next_payment_date_in_seconds);
-      history.SetString("adsNextPaymentDate",
-          base::TimeFormatWithPattern(next_payment_date, "MMMd"));
-    }
+void RewardsDOMHandler::OnGetTransactionHistory(
+    const bool success,
+    const double estimated_pending_rewards,
+    const uint64_t next_payment_date_in_seconds,
+    const uint64_t ad_notifications_received_this_month) {
+  if (!success) {
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_rewards_page_ui.cc:1466]: shouldn't we return this to front end so that FE can decide if to retry or something?

Comment (tmancey) [reply browser/ui/webui/brave_rewards_page_ui.cc:1466]: Retry logic is handled by the ads library where the UI is informed if the transaction history has changed, i.e. after initialization or when a user receives a reward

**Comment by tmancey** [comment]
CI failed due to unrelated browser test

**Review by NejcZdovc — APPROVED** [review]
rewards code looks good

File: vendor/brave-ios/Ads/BATBraveAds.mm [hunk 2]
```diff
@@ -437,7 +437,11 @@ - (void)updateAdRewards:(BOOL)shouldReconcile
 
 - (void)detailsForCurrentCycle:(void (^)(NSInteger adsReceived, double estimatedEarnings, NSDate *nextPaymentDate))completion
 {
-  ads->GetTransactionHistory(^(ads::StatementInfo list) {
+  ads->GetTransactionHistory(^(bool success, ads::StatementInfo list) {
+    if (!success) {
+      return;
```
Comment (kylehickinson) [inline vendor/brave-ios/Ads/BATBraveAds.mm:442]: Should call completion before returning here: (`completion(0, 0, nil)`)

Comment (tmancey) [reply vendor/brave-ios/Ads/BATBraveAds.mm:442]: If not successful ads is not initialized so we do not want to update the UI

Comment (kylehickinson) [reply vendor/brave-ios/Ads/BATBraveAds.mm:442]: The UI is already visible at this point, so either we implement an error state or we return with empty data

Comment (tmancey) [reply vendor/brave-ios/Ads/BATBraveAds.mm:443]: Fixed

**Comment by tmancey** [comment]
CI failed with unrelated browser tests

**Pull request merged**